docs/protocol/core/v3.0.rst

```diff
@@ -307,89 +331,128 @@ identify a data type.
+
 .. list-table:: Data types
    :header-rows: 1
 
    * - Identifier
      - Numerical type
      - Size (no. bytes)
      - Byte order
-   * - `bool`
-     - Boolean, with False encoded as `\\x00` and True encoded as `\\x01`
+   * - ``bool``
+     - Boolean, with False encoded as ``\\x00`` and True encoded as ``\\x01``
      - 1
      - None
-   * - `i1`
+   * - ``i1``
      - signed integer
      - 1
      - None
-   * - `<i2`
+   * - ``<i2``
      - signed integer
      - 2
      - little-endian
-   * - `<i4`
+   * - ``<i4``
      - signed integer
      - 4
      - little-endian
-   * - `<i8`
+   * - ``<i8``
      - signed integer
      - 8
      - little-endian
-   * - `>i2`
+   * - ``>i2``
      - signed integer
      - 2
      - big-endian
-   * - `>i4`
+   * - ``>i4``
      - signed integer
      - 4
      - big-endian
-   * - `>i8`
+   * - ``>i8``
      - signed integer
      - 8
      - big-endian
-   * - `u1`
+   * - ``u1``
      - signed integer
      - 1
      - None
-   * - `<u2`
+   * - ``<u2``
      - unsigned integer
      - 2
      - little-endian
-   * - `<u4`
+   * - ``<u4``
      - unsigned integer
      - 4
      - little-endian
-   * - `<u8`
+   * - ``<u8``
      - unsigned integer
      - 8
      - little-endian
-   * - `<f2`
+   * - ``<f2``
      - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
      - 2
      - little-endian
-   * - `<f4`
+   * - ``<f4``
      - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
      - 4
      - little-endian
-   * - `<f8`
+   * - ``<f8``
      - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
      - 8
      - little-endian
-   * - `>f2`
+   * - ``>f2``
      - half precision float: sign bit, 5 bits exponent, 10 bits mantissa
      - 2
      - big-endian
-   * - `>f4`
+   * - ``>f4``
      - single precision float: sign bit, 8 bits exponent, 23 bits mantissa
      - 4
      - big-endian
-   * - `>f8`
+   * - ``>f8``
      - double precision float: sign bit, 11 bits exponent, 52 bits mantissa
      - 8
      - big-endian
+   * - ``r*`` (Optional)
+     - raw bits,  use for extension type fallbacks
+     - variable, given by ``*``, is not limited to be a multiple of 8.
+     - N/A
+
 
 Floating point types correspond to basic binary interchange formats as
 defined by IEEE 754-2008.
 
+Additionally to these base types, an implementation should also handle the
+raw/opaque pass through type designated by the lowercase letter ``r`` followed
+by the number of bits. For example, ``r3``, ``r11`` should be understood as
+fallback types of respectively 3 and 11 bytes longs, and implementation should
+ensure to pass or accept those types unchanged when using extension fallback.
+For array whose total length is not a multiple of 8, zarr implementation can
+expect expects a buffer which length is a multiple of 8bits, padded with
+appropriate value at the end, and will in the same way return buffers padded to
+the next multiple of 8 bytes. 
```

Comment:
This paragraph confuses bits and bytes a lot. First it says bits, then it says bytes, then bits again, then bytes. I think they should all say bits.

Reply:
bits everywhere; sorry, thanks.